tests/test_api.rs

```diff
@@ -1726,6 +1726,57 @@ fn promise_reject_callback_no_value() {
   }
 }
 
+#[test]
+fn promise_hook() {
+  extern "C" fn hook(
+    type_: v8::PromiseHookType,
+    promise: v8::Local<v8::Promise>,
+    _parent: v8::Local<v8::Value>,
+  ) {
+    let scope = &mut unsafe { v8::CallbackScope::new(promise) };
+    let scope = &mut v8::HandleScope::new(scope);
+    let context = promise.creation_context(scope);
```

Comment:
Since you are going to call a function that is bound to `context`, I think you should also enter said context.
Otherwise the function now suddenly "sees" a different global object which can cause weird behavior.
Vice versa it may not be undesirable to leak the innards of the current context to the creation context.
So I'd suggest to use a ContextScope here.

Reply:
Actually nevermind - this is a test... 🤦

Reply:
I added a ContextScope although it doesn't really matter here as there's only one context. But it makes sense to be theoretically correct if the test suite also doubles as ersatz documentation.

Apropos the HandleScope, I see that pattern in several other tests.